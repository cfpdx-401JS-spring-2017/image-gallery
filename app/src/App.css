.App {
  text-align: center;
  display: flex;
  height: 100vh;
  width: 100%;
  justify-content: center;
  align-items: center;

}

.ThumbNail {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  height: 30vh;
}

.List {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

button {
  height: 10vh;
  width: 20vw;
  background-color: #476A6F;
  border: none;
  margin: 2vh 2vh;
  box-shadow: 1px 1px 1px 1px #202A25;
  border-radius: 10px;
}

.Link {
  color: #F7F0F0;
  text-decoration: none;
  font-size: 2em;
}

.Link:hover {
  
}
