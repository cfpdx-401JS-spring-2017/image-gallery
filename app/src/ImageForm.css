form {
    display: flex;
    flex-direction: column;
    width: 60vw;
    height: 50vh;
    background-color: #2F3E46;
    border-radius: 15px;
    color: #CAD2C5;
}

.submitButton {
    width: 10vw;
    margin: 0 auto;
    background-color: #CAD2C5;
    border: none;
    color: #2F3E46;
}

.InputField {
    width: 30vw;
    text-align: left;
    margin: 3vh 1vh;
    background-color: #CAD2C5;
    border: none;
    height: 5vh;
}