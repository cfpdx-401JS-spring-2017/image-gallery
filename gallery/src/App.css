.App {
  text-align: center;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.ThumbNail {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  height: 30vh;
}

.List {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

button {
  height: 8vh;
  width: 12vw;
  background-color: #2F3E46;
  color:#CAD2C5;
  border: none;
  margin: 2vh 2vh;
  box-shadow: none;
}
