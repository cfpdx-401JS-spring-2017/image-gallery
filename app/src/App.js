import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect
} from 'react-router-dom';

import Home from './components/Home';
import Albums from './components/albums/Albums';
import './styles/App.css';

class App extends Component {
  render() {
    return (
      <Router>
        <div className="App">
          <h2>CPDB: Cute Pomeranian Database</h2>
          <p>
            Welcome to the <code>Cute Pomeranian Database</code>! Check out our poms by clicking below:
          </p>
          <Switch>
            <Route exact path='/' component={Home}/>
            <Route path='/albums' component={Albums} />
            <Redirect to='/' />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
