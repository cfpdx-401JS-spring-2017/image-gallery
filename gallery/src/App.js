import React, { Component } from 'react';
import './App.css';
import AlbumsViewer from './AlbumsViewer';
import Albums from './Albums';

import {
  BrowserRouter as Router,
  Route, Switch, Redirect, Link
} from 'react-router-dom'

class App extends Component {
  constructor() {
    super();
    this.state = {
    }

  }

  render() {
  
    return (
      <div className="App">
        <Router>
            <div>
                <Switch>
                    {/*<Route exact path="/" component={App}/>*/}
                    <Route path="/albums/:id" component={AlbumsViewer}/>
                    <Route path="/albums" component={Albums}/>
                    <Redirect path="/albums" component={Albums}/>
                </Switch>
                <button>
                  <Link to="/albums">Go to Albums</Link>
                </button>
            </div>
        </Router>
      </div>  
    );
  }
}

export default App;
