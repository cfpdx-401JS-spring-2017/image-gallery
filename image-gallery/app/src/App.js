import React from 'react';
import './App.css';
import Header from './components/Header';
import Intro from './components/IntroView';
import Albums from './components/Albums'
import { BrowserRouter as Router, Route, Switch, Redirect } from 'react-router-dom';



export default function App() {   
    return (
      <Router>
        <div>
          <Header />

          <Switch>
            <Route exact path='/' component={Intro} />
            <Route path='/albums' component={Albums} />
            <Redirect to='/'/>
            
          </Switch>
        </div>
      </Router>

    );
  }


