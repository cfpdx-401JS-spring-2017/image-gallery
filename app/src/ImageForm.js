import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './ImageForm.css';

class ImageForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
        title: '',
        description: '',
        url: '',
    }

    this.titleHandler = this.titleHandler.bind(this);
  }

  titleHandler(e) {
      let {name, value} = e.target;
      this.setState({[name]: value});
  }

  render() {
      let {title, description, url} = this.state;
    return (
      <div className="ImageForm">
        <form onSubmit={ (e) => {
            e.preventDefault();
            this.props.addImage(this.state);
            }
        }>
            <label>
                Title:
                <input className="InputField" 
                    type="text" 
                    name="title" 
                    value={title}
                    onChange={this.titleHandler}
                />
            </label>
            <label>
                Description:
                <input className="InputField" 
                    type="text" 
                    name="description" 
                    value={description}
                    onChange={this.titleHandler}
                />
            </label>
            <label>
                Url:
                <input className="InputField" 
                    type="text" 
                    name="url" 
                    value={url}
                    onChange={this.titleHandler}
                />
            </label>
            <br />  
            <input className="submitButton" 
                type="submit" 
                value="Submit" 
            />
        </form>
      </div>  
    );
  }
}

ImageForm.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  url: PropTypes.string
};

export default ImageForm;