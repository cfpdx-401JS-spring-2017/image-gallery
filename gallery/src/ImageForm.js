import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './ImageForm.css';

class ImageForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
        title: '',
        desc: '',
        url: '',
    }

    this.titleHandler = this.titleHandler.bind(this);
  }

  titleHandler(e) {
      let {name, value} = e.target;
      this.setState({[name]: value});
  }

  render() {
      let {title, desc, url} = this.state;
    return (
      <div className="ImageForm">
        <form onSubmit={ (e) => {
            e.preventDefault();
            this.props.addImage(this.state);
            }
        }>
            <label>
                Title:
                <input className="InputField" 
                    type="text" 
                    name="title" 
                    value={title}
                    onChange={this.titleHandler}
                />
            </label>
            <label>
                Description:
                <input className="InputField" 
                    type="text" 
                    name="desc" 
                    value={desc}
                    onChange={this.titleHandler}
                />
            </label>
            <label>
                Url:
                <input className="InputField" 
                    type="text" 
                    name="url" 
                    value={url}
                    onChange={this.titleHandler}
                />
            </label>
            <br />  
            <input className="submitButton" 
                type="submit" 
                value="Submit" 
            />
        </form>
      </div>  
    );
  }
}

ImageForm.propTypes = {
  title: PropTypes.string,
  desc: PropTypes.string,
  url: PropTypes.string
};

export default ImageForm;