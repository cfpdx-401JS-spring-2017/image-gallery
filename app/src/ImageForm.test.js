import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import ImageForm from './ImageForm';

import renderer from 'react-test-renderer';
it('renders without crashing', () => {
    
  const div = document.createElement('div');
  ReactDOM.render(<ImageForm />, div);
});

it('renders a snapshot of main ImageForm component', () => {
  const component = renderer.create(<ImageForm  />);
  const bobo = component.toJSON();
  expect(bobo).toMatchSnapshot();
});