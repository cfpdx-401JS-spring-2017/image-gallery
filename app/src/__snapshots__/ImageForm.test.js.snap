// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of main ImageForm component 1`] = `
<div
  className="ImageForm"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Title:
      <input
        className="InputField"
        name="title"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <label>
      Description:
      <input
        className="InputField"
        name="description"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <label>
      Url:
      <input
        className="InputField"
        name="url"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <br />
    <input
      className="submitButton"
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
